Lib/ssl.py

```diff
@@ -1190,6 +1208,76 @@ def version(self):
         else:
             return None
 
+    def _ensure_wrapper(self):
+        if not self._sslobj:
+            raise ValueError("No SSL wrapper around " + str(self))
+
+    def key_update(self, updatetype):
+        self._ensure_wrapper()
+        self._sslobj.key_update(updatetype)
+
+    @property
+    def key_update_type(self):
+        self._ensure_wrapper()
+        return KeyUpdateTypes(self._sslobj.get_key_update_type())
+
+    def renegotiate(self, abbreviated=False):
+        self._ensure_wrapper()
+        if abbreviated:
+            self._sslobj.renegotiate_abbreviated()
+        else:
+            self._sslobj.renegotiate()
+
+    @property
+    def renegotiate_pending(self):
+        self._ensure_wrapper()
+        return self._sslobj.renegotiate_pending()
+
+
+for name, docstr in (
+        ('key_update', """\
+Schedule an update of the keys for the current TLS connection.
+
+If the updatetype parameter is set to KEY_UPDATE_NOT_REQUESTED then the
+sending keys for this connection will be updated and the peer will be
+informed of the change. If the updatetype parameter is set to
+KEY_UPDATE_REQUESTED then the sending keys for this connection will be
+updated and the peer will be informed of the change along with a
+request for the peer to additionally update its sending keys. It is an
+error if updatetype is set to KEY_UPDATE_NONE.
+
+key_update() must only be called after the initial handshake has been
+completed and TLSv1.3 has been negotiated. The key update will not take
+place until the next time an IO operation such as read() or write()
+takes place on the connection. Alternatively do_handshake()  can be
+called to force the update to take place immediately.
+
+Raises NotImplementedError if the TLS implementation doesn't support
+TLS 1.3.)
+
+:param updatetype: KeyUpdateTypes
```

Comment:
In Python we don't use sphinx markup in docstrings, documentation is not authogenerated but written manually.

Reply:
Ah, thanks for this one! I'll update docs too.

Reply:
This approach looks strange and uncommon to me. I opened https://github.com/python/cpython/pull/9972, which I believe is a better approach to handle common doc strings.

Reply:
Yes it is! I'll rebase up to #9972 once it is merged. Thanks!